src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```
@@ -0,0 +1,408 @@
+                if (IntPtr.Zero == contextPtr)
+                {
+                    throw CreateSslException("Failed to allocate SSL/TLS context");
+                }
+
+                libssl.SSL_CTX_ctrl(contextPtr, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
+
+                libssl.SSL_CTX_set_quiet_shutdown(contextPtr, 1);
+
+                if (certHandle != null && certKeyHandle != null)
+                {
+                    SetSslCertificate(contextPtr, certHandle, certKeyHandle);
+                }
+
+                sslContext.sslPtr = libssl.SSL_new(contextPtr);
+
+                libssl.SSL_CTX_free(contextPtr);     
+
+                if (IntPtr.Zero == sslContext.sslPtr)
+                {
+                    throw CreateSslException("Failed to create SSSL object from SSL context");
+                }      
+
+                IntPtr memMethodRead = libcrypto.BIO_s_mem();
+                IntPtr memMethodWrite = libcrypto.BIO_s_mem();
+                if ((IntPtr.Zero == memMethodWrite) || (IntPtr.Zero == memMethodRead))
+                {
+                    throw CreateSslException("Failed to return memory BIO method function");
+                }
+
+                sslContext.readBioPtr = libssl.BIO_new(memMethodRead);
+                sslContext.writeBioPtr = libssl.BIO_new(memMethodWrite);
+                if ((IntPtr.Zero == sslContext.readBioPtr) || (IntPtr.Zero == sslContext.writeBioPtr))
+                {
+                    throw CreateSslException("Failed to retun new BIO for a given method type");
+                }
+
+                if (isServer)
+                {
+                    libssl.SSL_set_accept_state(sslContext.sslPtr);
+                }
+                else
+                {
+                    libssl.SSL_set_connect_state(sslContext.sslPtr);
+                }
+
+                libssl.SSL_set_bio(sslContext.sslPtr, sslContext.readBioPtr, sslContext.writeBioPtr);
+            }
+            catch
```

We shouldn't be eating the exception here.